library.c

```diff
@@ -1869,6 +1883,36 @@ redis_sock_disconnect(RedisSock *redis_sock, int force TSRMLS_DC)
         if (redis_sock->persistent) {
             if (force) {
                 php_stream_pclose(redis_sock->stream);
+            } else {
+                zend_resource *le;
+                char host[1024], *persistent_id;
+                int persistent_len;
+                if (ZSTR_VAL(redis_sock->host)[0] == '/' && redis_sock->port < 1) {
+                    snprintf(host, sizeof(host), "unix://%s", ZSTR_VAL(redis_sock->host));
+                } else {
+                    const char *fmtstr = "%s:%d";
+#ifdef HAVE_IPV6
+                    if (strchr(ZSTR_VAL(redis_sock->host), ':') != NULL) {
+                        fmtstr = "[%s]:%d";
+                    }
+#endif
+                    snprintf(host, sizeof(host), fmtstr, ZSTR_VAL(redis_sock->host), redis_sock->port);
+                }
+                if (redis_sock->persistent_id) {
```

Comment:
I could be wrong, but I think with proper connection pooling there is no longer any need for directing the `persistent_id`, or by creating it with the timeout.  If we're separating connections in a pool we probably just want to hash by `host:port` or the fully qualified unix socket.

Reply:
Seems reasonable. I changed persistend_id for pool resource and remove it for sockets